common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/YubiKitCertBasedAuthManager.java

```diff
@@ -146,35 +193,61 @@ public void stopDiscovery() {
     public void requestDeviceSession(@NonNull final ISessionCallback callback) {
         final String methodTag = TAG + "requestDeviceSession:";
         synchronized (sDeviceLock) {
-            if (mDevice == null) {
+            if (isUsbDeviceConnected()) {
+                //Request a connection from mUsbDevice so that we can get a PivSession instance.
+                mUsbDevice.requestConnection(UsbSmartCardConnection.class, new Callback<Result<UsbSmartCardConnection, IOException>>() {
+                    @Override
+                    public void invoke(@NonNull final Result<UsbSmartCardConnection, IOException> value) {
+                        try {
+                            final SmartCardConnection c = value.getValue();
+                            final PivSession piv = new PivSession(c);
+                            final YubiKitSmartcardSession session = new YubiKitSmartcardSession(piv);
+                            callback.onGetSession(session);
+                        } catch (final Exception e) {
+                            callback.onException(e);
+                        }
+                    }
+                });
+            } else if (isNfcDeviceConnected()) {
```

Comment:
when a class big if-else like this all over the place, IMO it's a sign that it needs to be splitted.